app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -299,13 +285,34 @@ public function createOrder($reference, $session_quote_id = null)
 
         $this->validateSubmittedOrder($order, $quote);
 
-        // deactivate quote
-        $quote->setIsActive(false);
-        $quote->save();
-
         Mage::getModel('boltpay/payment')->handleOrderUpdate($order);
 
-        Mage::dispatchEvent('bolt_boltpay_save_order_after', array('order'=>$order, 'quote'=>$quote));
+        Mage::dispatchEvent('bolt_boltpay_save_order_after', array('order'=>$order, 'quote'=>$quote, 'transaction' => $transaction));
+
+        // Close out session by deactivating parent quote and deleting the immutable quote so that it can no
+        // longer be used.
+        /* @var Mage_Sales_Model_Quote $parentQuote */
+        $parentQuote = Mage::getModel('sales/quote')->loadByIdWithoutStore($quote->getParentQuoteId());
+        $parentQuote->setIsActive(false)
+            ->save();
+
+        if ($sessionQuoteId) {
+            $checkout_session = Mage::getSingleton('checkout/session');
+
+            $checkout_session
+                ->clearHelperData();
+
+            $checkout_session
+                ->setLastQuoteId($quote->getId())
+                ->setLastSuccessQuoteId($quote->getId());
+
+            // add order information to the session
+            $checkout_session->setLastOrderId($order->getId())
+                ->setRedirectUrl('')
+                ->setLastRealOrderId($order->getIncrementId());
+        }
+
+        $quote->delete();
```

Comment:
Just want to make sure that i'm following. We are deleting the immutable quote here, but is that the quote that is associated with the order? If so, shouldn't this be the one that we set to inactive and delete the parent quote? Otherwise the order will have reference to a quote id that doesn't exist.

Reply:
If the code reaches this point, then a Magento order has been created.  There is no longer a need for any quote, but, if there was, for some sort of auditing purposes, then the original quote theoretically should be the same.

For security purposes, once an immutable quote is used, it must no longer be available for any processing.  If it existed after its use, then if someone managed to get the transaction id or order token could attempt to reuse those to fool the system by maliciously reusing previous transactions.  Furthermore, on a more minor level, it is kept and set to inactive as to not use unnecessary space in the sales_flat_quote table.